.NET/Microsoft.Recognizers.Text.Number/Parsers/BaseCJKNumberParser.cs

```diff
@@ -571,22 +572,43 @@ private double GetIntValue(string intStr)
                         }
                         else
                         {
-                            beforeValue = Config.ZeroToNineMap[intStr[i]];
+                            double tmp = Config.ZeroToNineMap[intStr[i]];
+                            if (isPreDigit)
+                            {
+                                beforeValue = (beforeValue * 10) + tmp;
+                            }
+                            else
+                            {
+                                beforeValue = tmp;
+                            }
+
                             isRoundBefore = false;
                         }
                     }
                     else
                     {
-                        if (i == intStr.Length - 1 && (Config.CultureInfo.Name == "ja-JP" || Config.CultureInfo.Name == "ko-KR"))
+                        if ((Config.CultureInfo.Name == "ja-JP" || Config.CultureInfo.Name == "ko-KR") || char.IsDigit(intStr[i]))
```

Comment:
I'm not sure this new 'if' is correct. The previous behaviour would only enter here is both the first clause was true and the language was either JA or KO. Can you double check the behaviour and clean it up?

Reply:
OK, I will check it. Thank you.